File: app/src/main/java/com/yoesuv/infomadiun/utils/BottomNavigationViewBehavior.java
package com.yoesuv.infomadiun.utils;

import android.content.Context;
import android.support.design.widget.AppBarLayout;
import android.support.design.widget.BottomNavigationView;
import android.support.design.widget.CoordinatorLayout;
import android.util.AttributeSet;
import android.view.View;

/**
 *  Created by yusuf on 5/1/18.
 */

public class BottomNavigationViewBehavior extends CoordinatorLayout.Behavior<BottomNavigationView>{

    private int height;

    public BottomNavigationViewBehavior(Context context, AttributeSet attributeSet){
        super(context, attributeSet);
        this.height = AppHelper.getToolbarHeight(context);
    }

    @Override
    public boolean layoutDependsOn(CoordinatorLayout parent, BottomNavigationView child, View dependency) {
        return dependency instanceof AppBarLayout;
    }

    @Override
    public boolean onDependentViewChanged(CoordinatorLayout parent, BottomNavigationView child, View dependency) {
        if(dependency instanceof AppBarLayout){
            CoordinatorLayout.LayoutParams lp = (CoordinatorLayout.LayoutParams) child.getLayoutParams();
            int bottomMargin = lp.bottomMargin;
            int distanceToScroll = child.getHeight() + bottomMargin;
            float ratio = dependency.getY()/ height;
            child.setTranslationY(-distanceToScroll * ratio);
        }
        return true;
    }
}

File: app/src/main/java/com/yoesuv/infomadiun/utils/BottomNavigationViewBehavior.kt
package com.yoesuv.infomadiun.utils

import android.content.Context
import com.google.android.material.appbar.AppBarLayout
import com.google.android.material.bottomnavigation.BottomNavigationView
import androidx.coordinatorlayout.widget.CoordinatorLayout
import android.util.AttributeSet
import android.view.View

/**
 *  Created by yusuf on 5/1/18.
 */

class BottomNavigationViewBehavior(context: Context, attributeSet: AttributeSet) : androidx.coordinatorlayout.widget.CoordinatorLayout.Behavior<BottomNavigationView>(context, attributeSet) {

    private val height: Int = AppHelper.getToolbarHeight(context)

    override fun layoutDependsOn(parent: androidx.coordinatorlayout.widget.CoordinatorLayout, child: BottomNavigationView, dependency: View): Boolean {
        return dependency is AppBarLayout
    }

    override fun onDependentViewChanged(parent: androidx.coordinatorlayout.widget.CoordinatorLayout, child: BottomNavigationView, dependency: View): Boolean {
        if (dependency is AppBarLayout) {
            val lp = child.layoutParams as androidx.coordinatorlayout.widget.CoordinatorLayout.LayoutParams
            val bottomMargin = lp.bottomMargin
            val distanceToScroll = child.height + bottomMargin
            val ratio = dependency.y / height
            child.translationY = -distanceToScroll * ratio
        }
        return true
    }
}
